### PR TITLE
feat(web): my-transactions section, style preview component, pass all variables, modularize, refactors

### PR DESCRIPTION
**Pull request by kemuru**

### CODE REVIEW

File: web/src/components/Modal/ProposeSettlementModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { StyledModal } from "../StyledModal";
+
+interface IProposeSettlementModal {
+  toggleModal: () => void;
+  text: string;
+}
+
+const ProposeSettlementModal: React.FC<IProposeSettlementModal> = ({ toggleModal, text }) => {
+  const containerRef = useRef(null);
+  useFocusOutside(containerRef, () => {
```
[the remark this answers is not in the document]
Comment (kemuru) [reply web/src/components/Modal/ProposeSettlementModal/index.tsx:17]: yep, it works

[the remark this answers is not in the document]
Comment (kemuru) [reply web/src/components/Modal/ProposeSettlementModal/index.tsx:17]: wait, im seeing some side effects

[the remark this answers is not in the document]
Comment (kemuru) [reply web/src/components/Modal/ProposeSettlementModal/index.tsx:17]: ok no, its good its good

[the remark this answers is not in the document]
Comment (alcercu) [reply web/src/components/Modal/ProposeSettlementModal/index.tsx]: then, adding it as V2 backlog to substitute it there too 🙌🏻

**Review by alcercu — APPROVED** [review]
lgtm

**Pull request merged**